lib/std/uuid.zig

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2020 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+const rand = @import("std.zig").rand;
+
+const Self = @This();
+id: u128,
+
+/// Creates a brand new UUID v4 object given a random number generator.
+pub fn newv4(gen: *rand.Random) Self {
+    //This sets the 4 bits at the 48 bit to 0
+    const flip: u128 = (0b1111 << 48);
+
+    //We then set the 4 bits at 48 to 0x4 (0b0100)
+    return Self{ .id = gen.int(u128) & ~flip | 0x4 << 48 };
+}
+
+/// Creates a string from the ID.
+/// Format: xxxxxxxx-xxxx-4xxx-xxxx-xxxxxxxxxxxx
+pub fn format(self: *Self) []const u8 {
```

Comment:
Check [this](https://github.com/ziglang/zig/blob/master/lib/std/fmt.zig#L77,L82) out, `format` follows a well defined protocol.